tests/parser/features/external_contracts/test_external_contract_calls.py

```diff
@@ -2452,3 +2452,74 @@ def do_stuff(f: Foo) -> uint256:
     c2 = get_contract(callee_code)
 
     assert c1.do_stuff(c2.address) == 1
+
+
+def test_calldata_clamp(get_contract, assert_tx_failed):
```

Comment:
also need a test for calldata exceeding the expected length in the case of static data.

side note -- i think these tests might be clearer if the malformed data is prepared in python.